lib/plugins/milestones.js

```diff
@@ -0,0 +1,37 @@
+const Diffable = require('./diffable')
+
+module.exports = class Milestones extends Diffable {
+  find () {
+    let allMilestones = []
+
+    allMilestones.push(this.github.issues.getMilestones(Object.assign({}, {state: 'open'}, this.repo)).then(res => res.data))
+    allMilestones.push(this.github.issues.getMilestones(Object.assign({}, {state: 'closed'}, this.repo)).then(res => res.data))
+
+    return Promise.all(allMilestones)
+      .then(milestones => {
+        return milestones[0].concat(milestones[1])
+      })
+  }
+
+  comparator (existing, attrs) {
+    return existing.title === attrs.title
+  }
+
+  changed (existing, attrs) {
+    return existing.description !== attrs.description || existing.due_on !== attrs.due_on
+  }
+
+  update (existing, attrs) {
+    return this.github.issues.updateMilestone(Object.assign({}, attrs, this.repo, { number: existing.number }))
```

Comment:
Minor nit here. You can replace the empty object at the beginning of `Object.assign` with the last argument. In other places an empty object is used as the first arg to create a new object and avoid mutating an existing object that was passed in as a param. But since a new object is being constructed here either way, it can be used.

```diff
- Object.assign({}, attrs, this.repo, { number: existing.number })
+ Object.assign({ number: existing.number }, attrs, this.repo)
```